lighthouse-core/computed/image-records.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('./computed-artifact.js');
+
+class ImageRecords {
+  /**
+   * @param {LH.Artifacts.NetworkRequest[]} networkRecords
+   */
+  static indexNetworkRecords(networkRecords) {
+    return networkRecords.reduce((map, record) => {
+      // An image response in newer formats is sometimes incorrectly marked as "application/octet-stream",
+      // so respect the extension too.
+      const isImage = /^image/.test(record.mimeType) || /\.(avif|webp)$/i.test(record.url);
```

Comment:
Do we need this anymore?

Reply:
I would say until proven otherwise, yes. AVIF is still unknown enough that I suspect many older servers will serve it as `application/octet-stream` for quite some time.